PelicanPlatform/pelican
# Refactor origin_ui to web_ui to reuse basic logic for server web UI/API

Closes #308 

- This PR also introduced basic username/password auth and its set of APIs to all servers, as well as common APIs such as `/config`, `/health` to all servers
- Changed the auth URLs from `/api/v1.0/origin_ui` to `/api/v1.0/auth/`, the origin UI code is updated to the correct URLs
- Changed the name for one-time activation code and web ui password file to `server-web-activation-code` and `server-web-passwd` respectively.
- Put `/health` API behind the authentication middleware

## Testing Instructions
* Spin up your `director` `registry` and `origin`
* Go to `https://localhost:8444/view` and ensure you can visit origin's web ui as before
* Close all servers and remove `origin-ui-passwd` under your `/etc/pelican` directory
* Restart any of the server and you should be able to see that the terminal shows the one-time passcode to initialize the UI, although you can only do it through origin UI now. 
* Go to the terminal where origin runs, find the one-time code and go to the origin's initialization page, activate UI using the code, and set your new admin password.
* Under your `/etc/pelican` now you should see a `server-web-passwd`
